### PR TITLE
add pdftopng

### PR DESCRIPTION
**Pull request by izahn**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by izahn** [comment]
@conda-forge-admin please restart ci

**Comment by izahn** [comment]
@conda-forge-admin please restart ci

**Comment by izahn** [comment]
Ready for review @conda-forge/help-python-c 

I got it to build, but seems like kind of a mess, required patching `setup.py`. Happy for any suggestions!

File: recipes/pdftopng/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - click
+    - pip
+  run:
+    - python
+    - poppler
+    - click
+
+test:
+  imports:
+    pdftopng
```
Comment (chrisburr) [inline recipes/pdftopng/meta.yaml:40]:
```suggestion
    - pdftopng
```

**Comment by izahn** [comment]
@vinayak-mehta do you want to co-maintain this one? It is needed by `camelot-py >=0.10`.

**Comment by izahn** [comment]
@vinayak-mehta just checking one last time to see if you want to be a co-maintainer here.

**Comment by izahn** [comment]
This one is ready for review and merge review @conda-forge/help-python-c

**Comment by vinayak-mehta** [comment]
@izahn Sorry for the super late reply. I haven't been able to spend time on the projects due to other obligations, but I'm planning to get back into it slowly in the coming weeks. 

> I got it to build, but seems like kind of a mess, required patching setup.py. Happy for any suggestions!

Thanks for working on this PR! Let me know if I can do anything to help speed up your work, I can look into it this weekend.

**Comment by izahn** [comment]
No worries @vinayak-mehta thanks in advance for taking a look! The main thing is just that it seemed like a lot of things were hard-coded in `setup.py` and I had to [do some patching](https://github.com/izahn/staged-recipes/blob/pdftopng/recipes/pdftopng/0001-fix-setup.patch) to point things to the conda-forge locations. It should be OK I think, but maybe there are some improvements that can be made.

I added you as a maintainer, and invited you to collaborate over in https://github.com/izahn/staged-recipes so you can push directly to the `pdftopng` branch there if you like. Thanks again!

**Comment by izahn** [comment]
Ready for review and merge @conda-forge/help-python-c

File: recipes/pdftopng/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - setuptools
+    - cython
+    - pybind11
+    - poppler
+    - click
+    - pip
+  run:
+    - python
+    - poppler
+    - click
```
Comment (carterbox) [inline recipes/pdftopng/meta.yaml:36]: Is there a console entry point for this package? You need to register entry points with conda. Also please add a call to the entry point in the tests section.

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

File: recipes/pdftopng/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - pybind11                               # [build_platform != target_platform]
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - setuptools
+    - cython
+    - pybind11
+    - poppler
+    - click
```
Comment (carterbox) [inline recipes/pdftopng/meta.yaml:31]: Click is python-only, so it should not need to be in the build/host section.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/pdftopng/meta.yaml (hunk 3 above)
Comment (izahn) [reply recipes/pdftopng/meta.yaml]: Done, thank you!

File: recipes/pdftopng/meta.yaml (hunk 2 above)
Comment (izahn) [reply recipes/pdftopng/meta.yaml:35]: Hmm, is this really needed? The entry point is already specified in https://github.com/vinayak-mehta/pdftopng/blob/633c594f24e77f00f00f0675b49daea2c71e279b/setup.py#L171 and the package produces the appropriate scripts/binaries. Do we really need to specify it in `meta.yaml` also?

Comment (izahn) [reply recipes/pdftopng/meta.yaml:35]: I see https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=entry_point which suggests that at least a redundant entry point in `meta.yaml` won't hurt anything, so I've gone ahead and added it as requested.

**Comment by izahn** [comment]
I think this one is ready @conda-forge/help-python-c

File: recipes/pdftopng/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipes/pdftopng/meta.yaml:37]: They are needed (especially for `noarch: python` packages)

**Pull request merged**